src/libman/Contracts/HostInteraction.cs

```diff
@@ -0,0 +1,120 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Web.LibraryManager.Contracts;
+
```

Comment:
Why IHostInteration interface was not enough for the CLI. If you need more data you can add to that interface. Also, we should implement IHostInteration at your project level and not in the contracts. For now, just reuse the one we have for .Build project so we are consistent. I am planning on creating a base HostInteration Class that implements common behavior among all our "in box" extensions (.Build, .Vsix, .ClI) implementations (e.g.: Write/Read to files).

Reply:
The reason for adding the `IHostInteractionsInternal` was to not have force others to implement the `UpdateWorkingDirectory()` 
I did intend to move the `DeleteFiles` to the `IHostInteractions` interface, bu the UpdateWorkingDirectory doesn't seem a useful thing in there.